BLSQ/iaso
# IA-2809: Periods: add possibility to select "No Period" on the Form interface

Explain what problem this PR is resolving
- Periods: add possibility to select "No Period" on the Form interface
Related JIRA tickets : [IA-2809](https://bluesquare.atlassian.net/browse/IA-2809)

## Self proofreading checklist


## Changes
- Add migration to alter iaso_form table on single_per_period, periods_before_allowed and periods_after_allowed field to be nullable
- Add NO_PERIOD in the period types
- Hide all related periods fields(single_per_period, periods_before_allowed and periods_after_allowed) when NO_PERIOD type is selected
## How to test
- Create a new form or update one
- Change period type and check if it works well
## Print screen / video
[Screencast from 2024-04-04 13-01-12.webm](https://github.com/BLSQ/iaso/assets/19631540/9f26da03-d1bb-410f-9602-8f39abaa0fb2)


[IA-2809]: https://bluesquare.atlassian.net/browse/IA-2809?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
